### ISSUE TITLE
Font rendered as white boxes

### ISSUE THREAD

**Issue by j-jorge** [axmolengine/axmol#2003]
Since commit 74612ef29662b4be72ebb50f9234b4d519fa992e the labels rendered without effect are incorrectly displayed.

- axmol version: 2.1.3
- devices test on: Android
- developing environments
   - NDK version: r26
   - cmake version: .28.3
Steps to Reproduce:

1. Create a label with no effect:

```c++
ax::TTFConfig ttf_config;

ttf_config.fontFilePath = "some-font.ttf";
ttf_config.fontSize = 12;
ttf_config.outlineSize = 0;
ttf_config.italics = false;
ttf_config.bold = false;
ttf_config.underline = false;
ttf_config.strikethrough = false;

ax::TextHAlignment horizontal_align = ax::TextHAlignment::LEFT;

ax::Label* label =  ax::Label::createWithTTF(ttf_config, "…", horizontal_align);
```

2. Display the label:
![image](https://github.com/axmolengine/axmol/assets/2023446/0426003b-2a4b-410a-97a9-ff16e51bd31d)

Note that the labels are displayed correctly if I set an outline.

Regarding the problematic commit, I guess the following is also important here: I build the app with `AX_GLES_PROFILE=200` and link with GLESv2.

**Comment by halx99** [comment]
Download cpp-test  apk https://github.com/axmolengine/axmol/actions/runs/9623334152/artifacts/1627192539 and run on your device check `31:Node: Label -New API/32`, does it have same issue on your device?

on my device, works fine

![image](https://github.com/axmolengine/axmol/assets/6977319/20d76022-49ff-482a-87df-1ff24862ef72)

**Comment by j-jorge** [comment]
This one works as expected. Was there a fix between the faulty commit and this build? or is it a problem on my side?

**Comment by j-jorge** [comment]
I still have the issue on the current dev branch, so I guess the problem is in my compilation flags.

**Comment by halx99** [comment]
not sure, by default the github ci also build apk with `AX_GLES_PROFILE=200`

**Comment by halx99** [comment]
I can't reproduce with your code

**Comment by j-jorge** [comment]
I am closing this issue since it seems to be an issue on my side. I'll come back to it if I find something useful for Axmol. Thanks.

**Comment by smilediver** [comment]
Any errors in console? Try checking what kind of texture is being used for that configuration (is it simple alpha mask or SDF?) and which shader. And then check if the shader matches texture type, and if the final shader in the apk is correct.

Also, does your project build and run on other platforms? Could you check if those work correctly?

**Comment by j-jorge** [comment]
There are no error in the logs and the shader seems to be correct (label_normal.frag and positionTextureColor.vert). I don't get what could be wrong. If I set a transparent outline to use label_outline.frag then the rendering is ok, yet the code from this shader, when effectType == 0, is almost exactly the code from the former.

I managed to get the correct rendering by patching [line 1299 of Label.cpp](https://github.com/axmolengine/axmol/blob/d79a202e6e3c4410af1bd48068c1e2eed161418d/core/2d/Label.cpp#L1290), changing the last argument from `true` to `false`.

```c++
setFontAtlas(newAtlas, _fontConfig.distanceFieldEnabled, /* useA8Shader = */ false);
```

I don't know why there is no problem in CppTests though. Anyway, since the intent of commit 74612ef29662b4be72ebb50f9234b4d519fa992e is to remove deprecated pixel formats, including A8, does it make sense to keep `Label::_useA8Shader`?
